### PR TITLE
FIX: ISXB-226 - Feature flag for platform IOCTL setting.

### PR DESCRIPTION
**Pull request by ekcoh**
FIX: Fixed an issue where setting an internal feature flag using IOCTL to communicate with native backend to `false` after being `true` would not have any effect when attempting to change gamepad backend (case ISXB-226).

### Description

The existing feature flag to switch between Windows.Gaming.Input and XInput as gaming backend on Windows didn't have any  effect when switching from on (true) to off (false).

### Changes made

Change so that feature flag is tri-state, either {unset/unspecified/default, off, on}
Modified InputManager to not send IOCTL if unspecified since backend defines default setting.
Added a menu extension so this option can be toggled from QA menu.

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Review by jimon — APPROVED** [review]

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ however, it has to be formatted properly to pass verification tests.
 - Fix UI sometimes ignoring the first mouse click event after losing and regaining focus ([case ISXB-127](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-127).
 - Fixed issue when using MultiplayerEventSystems where the visual state of UI controls would change due to constant toggling of CanvasGroup.interactable on and off ([case ISXB-112](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-112)).
 - Fixed an issue where the Input Action asset icon would not be visible during asset creation ([case ISXB-6](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-6)).
-
+- Fixed an issue where setting an internal feature flag using IOCTL to communicate with native backend to `false` after being `true` would not have any effect (case ISXB-226).
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Should it have the issuetracker link here?

**Review by jamesmcgill — APPROVED** [review]

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: The issue isn't public so there is none from what I can see

**Comment by ekcoh** [comment]
@jimon Leaving this one open until we have explored player settings path in case we do not need it.

**Comment by ekcoh** [comment]
@jamesmcgill @jimon Updated this branch based on changes introduced here: https://github.cds.internal.unity3d.com/unity/unity/commit/15687808993f07d8d77aad15b617163f99458e60

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1553) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1553) before we can accept your contribution.<br/><hr/>**Dmytro Ivanov** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/InputSystem?pullRequest=1553) it.</sub>

**Comment by ekcoh** [comment]
Closed this since obsolete after switching to settings-based WGI opt-in.

**Pull request closed without merging**